spec/system/consumer/split_checkout_spec.rb

```diff
@@ -207,10 +207,111 @@
         end
       end
 
-      describe "filling out delivery details" do
+      describe "filling out order details" do
         before do
           fill_out_details
           fill_out_billing_address
+          order.update(user_id: user.id)
+        end
+
+        context "billing address" do
+          describe "checking the default address box" do
+            before do
+              check "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            it "before saving has no default address" do
+              expect(user.bill_address).to be_nil
+              expect(order.bill_address).to be_nil
+            end
+
+            context "proceeding to payment" do
+              before do
+                proceed_to_payment
+              end
+
+              it "creates a new default bill address" do
+                expect(order.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(order.customer.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(user.reload.bill_address.address1).to eq "Rue de la Vie, 77"
```

Comment:
You could even do this:
```suggestion
                expect {
                  proceed_to_payment
                  order.reload
                  user.reload
                }.to change {
                  order.bill_address&.address1.from(nil).to("Rue de la Vie, 77")
                }.and change {
                  order.customer.bill_address&.address1.from(nil).to("Rue de la Vie, 77")
                }.and change {
                  order.bill_address&.address1.from(nil).to("Rue de la Vie, 77")
                }
```

Reply:
It makes even _more_ sense to remove the `it` block (previous comment) by seeing here that we can include the previous state `(nil)` and compare it to the changed state `("Rue...")`. Cool. Thanks!! :sparkles: